src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -893,103 +1141,125 @@ private BoundExpression GetDefaultParameterValue(SyntaxNode syntax, ParameterSym
+                                     
+                BoundExpression lineLiteral = MakeLiteral(syntax, ConstantValue.Create(line), compilation.GetSpecialType(SpecialType.System_Int32), localRewriter);
 
                 if (parameterType.IsNullableType())
                 {
-                    defaultValue = MakeConversionNode(lineLiteral, parameterType.GetNullableUnderlyingType(), false);
+                    TypeSymbol nullableType = parameterType.GetNullableUnderlyingType();
+                    defaultValue = isLowering
+                        ? localRewriter.MakeConversionNode(lineLiteral, nullableType, false)                
+                        : MakeConversionForIOperation(lineLiteral, nullableType, syntax, compilation, diagnostics, false); 
 
                     // wrap it in a nullable ctor.
                     defaultValue = new BoundObjectCreationExpression(
                         syntax,
-                        UnsafeGetNullableMethod(syntax, parameterType, SpecialMember.System_Nullable_T__ctor),
+                        UnsafeGetNullableMethod(syntax, parameterType, SpecialMember.System_Nullable_T__ctor, compilation, diagnostics),
+                        null,
                         defaultValue);
                 }
                 else
-                {
-                    defaultValue = MakeConversionNode(lineLiteral, parameterType, false);
-                }
+                {                                                                                                                                     
+                    defaultValue = isLowering
+                        ? localRewriter.MakeConversionNode(lineLiteral, parameterType, false)
+                        : MakeConversionForIOperation(lineLiteral, parameterType, syntax, compilation, diagnostics, false);
+                }  
             }
             else if (parameter.IsCallerFilePath && ((callerSourceLocation = GetCallerLocation(syntax, enableCallerInfo)) != null))
             {
-                string path = callerSourceLocation.SourceTree.GetDisplayPath(callerSourceLocation.SourceSpan, _compilation.Options.SourceReferenceResolver);
-                BoundExpression memberNameLiteral = MakeLiteral(syntax, ConstantValue.Create(path), _compilation.GetSpecialType(SpecialType.System_String));
-                defaultValue = MakeConversionNode(memberNameLiteral, parameterType, false);
+                string path = callerSourceLocation.SourceTree.GetDisplayPath(callerSourceLocation.SourceSpan, compilation.Options.SourceReferenceResolver);
+                BoundExpression memberNameLiteral = MakeLiteral(syntax, ConstantValue.Create(path), compilation.GetSpecialType(SpecialType.System_String), localRewriter); 
+                defaultValue = isLowering
+                    ? localRewriter.MakeConversionNode(memberNameLiteral, parameterType, false)
```

Comment:
>? localRewriter.MakeConversionNode(memberNameLiteral, parameterType, false) [](start = 20, length = 75)

The pattern ```isLowering ? localRewriter.MakeConversionNode(...) : MakeConversionForIOperation(...)```  is used in several places, consider extracting it into a helper method.